.dialyzer_ignore.exs

```diff
@@ -5,5 +5,7 @@
   # Following warnings are due to bad implementation of riverside.
   {"lib/phoenix/router.ex", :pattern_match, 402},
   {"lib/riverside.ex", :pattern_match, 876},
-  {"lib/ae_mdw_web/websocket/socket_handler.ex", :overlapping_contract, 2}
+  {"lib/ae_mdw_web/websocket/socket_handler.ex", :overlapping_contract, 2},
+  {"lib/ae_mdw/db/mutations/contract_call_mutation.ex", :pattern_match, 58},
+  {"lib/ae_mdw/db/mutations/contract_call_mutation.ex", :pattern_match, 59}
```

Comment:
What specific errors do you get here? I'd like to avoid ignoring type errors

Reply:
dialyzer is not following the short circuit condition on:
`Contract.is_successful_call?(fun_arg_res) and not Contract.is_non_stateful_aex9_function?(fun_arg_res.function)`
it complains because `fun_arg_res` is of the type `fun_arg_res_or_error()` (when a contract fails it's an error tuple) but fun_arg_res.function is a safe access because of `Contract.is_successful_call?`.

I've discarded the other way doing the parttern mathing of the whole fun_args_res_error that because it looses orthogonality. The function would become:
`Contract.is_non_stateful_aex9_successful_call?`

Reply:
would you have another idea?